src/VerifyEmailHelper.php

```diff
@@ -45,6 +46,11 @@ public function __construct(UrlGeneratorInterface $router, /* no typehint for BC
         $this->queryUtility = $queryUtility;
         $this->tokenGenerator = $generator;
         $this->lifetime = $lifetime;
+
+        if (!$uriSigner instanceof UriSigner) {
+            /** @psalm-suppress UndefinedFunction */
+            @trigger_deprecation('symfonycasts/verify-email-bundle', '1.17.0', 'Not providing an instance of %s is deprecated. It will be required in v2.0', UriSigner::class);
+        }
```

Comment:
We should also deprecated "passing $queryUtility as 3rd argument is deprecated. The arg will be removed in the future..." but usually when we do this - the param is `= null`able. We can't do that here because the last 2 arg's are _not_ nullable. 

Do we let this ride as is and document this "non-deprecated" change in `Upgrade.md` or add the deprecation anyway without providing a way for the user to "silence" it?

Reply:
Hmm, maybe we lave the argument as is & not deprecated for now. Then we deprecate it in v2, once it's fully unused. There may be a better way to handle this, but I can't see it at the moment.